### PR TITLE
Made output of error more specific to bad parameter

### PR DESCRIPTION
**Pull request by oatmealraisin**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1199391
Redo of botched pull request https://github.com/openshift/origin/pull/9237
Nothing has changed from that pull
@bparees

### CODE REVIEW

**Comment by bparees** [comment]
can you give me a sample of the new error output?

**Comment by oatmealraisin** [comment]
New error: 

the template "ruby-helloworld-sample": Template "ruby-helloworld-sample" is invalid: template.parameters[0].Generate: Invalid value: "test": template.parameters[0]: Invalid 'test' generator for parameter ADMIN_USERNAME
Maybe truncate some of the duplicate information?

**Comment by bparees** [comment]
yeah, just
```
the template "ruby-helloworld-sample": Template "ruby-helloworld-sample" is invalid: template.parameters[0].Generate: Invalid value: "test": Unknown generator name 'test' for parameter ADMIN_USERNAME
```

would be good.

File: pkg/template/template.go [hunk 1]
```diff
@@ -163,7 +163,8 @@ func (p *Processor) GenerateParameterValues(t *api.Template) *field.Error {
 		if param.Generate != "" {
 			generator, ok := p.Generators[param.Generate]
 			if !ok {
-				return field.NotFound(templatePath, param)
+				err := fmt.Errorf("Uknown generator name '%v' for parameter %s", param.Generate, param.Name)
```
Comment (bparees) [inline pkg/template/template.go]: typo in Unknown

**Comment by bparees** [comment]
@oatmealraisin bump

File: pkg/template/template.go (hunk 1 above)
Comment (oatmealraisin) [inline pkg/template/template.go]: fixed

**Comment by bparees** [comment]
@oatmealraisin please squash your commits and then lgtm.

**Comment by oatmealraisin** [comment]
@csrwng ptal?

**Comment by csrwng** [comment]
LGTM [merge]

**Comment by csrwng** [comment]
@oatmealraisin The TestParameterGenerators unit test at pkg/template/template_test.go is failing because you changed the type of error for the field, pls fix the test.

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
@oatmealraisin looks like the unit test is still broken, please let me know if you need help running it locally
```
	template_test.go:169: test[2]: Unexpected error type: Expected: template.parameters[0], got template.parameters[0].Generate
```

Removed merge tag for now.

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**